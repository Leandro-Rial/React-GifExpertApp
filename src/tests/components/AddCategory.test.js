import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe('Prueba AddCategory', () => {

    const setCategories = jest.fn();

    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />)
    });

    test('debe de mostar <AddCategory /> correctamente', () => {
        
        expect(wrapper).toMatchSnapshot()

    });

    test('debe de cambiar la caja de textto', () => {
        
        const inputAddCategory = wrapper.find("input");

        const value = "Helouda";

        inputAddCategory.simulate("change", { target: { value } })
        
        expect(wrapper.find("p").text().trim()).toBe( value )

    });

    test('no debe de postear la informacion con submit', () => {
        
        wrapper.find("form").simulate("submit", { preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled()

    });

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        // simular el Change
        const value = "Helouda";

        wrapper.find("input").simulate("change", { target: { value } })

        
        // simular el Submit
        wrapper.find("form").simulate("submit", { preventDefault(){} })
        
        
        // SetCategories debe ser llamado
        expect( setCategories ).toHaveBeenCalled()
        

        // InputValue clean
        const inputValue = wrapper.find("input").prop("value")
        
        expect( inputValue ).toBe("")
        
    });
});