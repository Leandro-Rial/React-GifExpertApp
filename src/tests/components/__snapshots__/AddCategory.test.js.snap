// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba AddCategory debe de mostar <AddCategory /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
